inject_test.go

```diff
@@ -269,11 +272,32 @@ func TestInject(t *testing.T) {
 				T2 *type2 `optional:"true"`
 			}
 		}
-		require.NoError(t, app.Start(context.Background(), Inject(&out)),
-			"failed to start")
 
+		app := New(
+			Provide(new1),
+			Inject(&out),
+		)
+
+		require.NoError(t, app.Start(context.Background()), "failed to start")
 		assert.NotNil(t, out.Result.T1, "T1 must not be nil")
 		assert.Nil(t, out.Result.T2, "T2 must be nil")
 		assert.True(t, gave1 == out.Result.T1, "T1 must match")
 	})
 }
+
+func ExampleInject() {
```

Comment:
Yep! 
![screen shot 2017-06-16 at 11 00 55 am](https://user-images.githubusercontent.com/972790/27238800-1d4c0ac8-5283-11e7-951c-22640f71fd45.png)

See https://blog.golang.org/examples for details.